eclair-core/src/main/scala/fr/acinq/eclair/crypto/Sphinx.scala

```diff
@@ -36,35 +32,13 @@ import scala.util.{Failure, Success, Try}
   * see https://github.com/lightningnetwork/lightning-rfc/blob/master/04-onion-routing.md
   */
 object Sphinx extends Logging {
-  val Version = 0.toByte
 
-  // length of a MAC
+  // We use HMAC-SHA256 which returns 32-bytes message authentication codes.
   val MacLength = 32
 
-  // length of a payload: 33 bytes (1 bytes for realm, 32 bytes for a realm-specific packet)
-  val PayloadLength = 33
-
-  // max number of hops
-  val MaxHops = 20
-
-  // onion packet length
-  val PacketLength = 1 + 33 + MacLength + MaxHops * (PayloadLength + MacLength)
+  def mac(key: ByteVector, message: ByteVector): ByteVector32 = Mac.hmac256(key, message)
```

Comment:
Premature optimization :D
To avoid creating objects all the time (creating more garbage) when I could just call the method directly.
But if you prefer I change that.

Reply:
Well I saw that you were using it somewhere else. Otherwise I was wondering why introduce this abstraction in the first place. I think you can keep your optimization and still use the interface.

Reply:
The interface is necessary for the `prependmac` codec, which is why I took some code out of Sphinx to the Mac file so that codecs can use it.